include/bx/platform.h

```diff
@@ -6,6 +6,10 @@
 #ifndef BX_PLATFORM_H_HEADER_GUARD
 #define BX_PLATFORM_H_HEADER_GUARD
 
+#ifdef __APPLE__
+#include <TargetConditionals.h>
```

Comment:
No platform includes... You have to figure out how to detect it without it. If you didn't notice there are no other platform includes.

Reply:
I've replaced platform include with `#elif __has_builtin(__is_target_os) && __is_target_os(xros)` as there is no `__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__` equivalent for visionOS

Reply:
They have these for all platforms except visionOS?
```cpp
__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__
__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__
__ENVIRONMENT_TV_OS_VERSION_MIN_REQUIRED__
__ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__
```

> Do you think that's okay?

This is fine, but you have to guard first `__has_builtin` with something, since it's not available everywhere:
```cpp
#elif __has_builtin(__is_target_os) && __is_target_os(xros)
```

For example (not sure if this will work, just a guess):

```cpp
#elif defined(__has_builtin) && __has_builtin(__is_target_os) && __is_target_os(xros)
```

Reply:
Yes, unfortunately this is missing for visionOS.. 

Apple left some `TODO` comments in `AvailabilityInternal.h` (header where the rest of the defines are mentioned) but there is no  `__ENVIRONMENT_XR_OS_VERSION_MIN_REQUIRED__` or `__ENVIRONMENT_VISIONOS_OS_VERSION_MIN_REQUIRED__`. 

![CleanShot 2024-04-29 at 15 38 19@2x](https://github.com/bkaradzic/bx/assets/52801365/f0b2bfb4-2aa7-4652-ad7e-36c100f0d061)

I've checked and `#elif defined(__has_builtin) && __has_builtin(__is_target_os) && __is_target_os(xros)` works great.

Reply:
> I've checked and #elif defined(__has_builtin) && __has_builtin(__is_target_os) && __is_target_os(xros) works great.

Great! LGTM.

Are you still adding changes or this is it? I'm ready to merge it.

Reply:
From the point of bx repo I think it's all. If anything pops out I can create a follow up PR. 

So yeah, I think we can merge it 😄